Add in JNI APIs for scan, replace_nulls, group_by.scan, and group_by.replace_nulls [skip ci]

revans2 opened the pull request:
To be able to do a running window test prototype I added in APIs for `scan`, `group_by.scan`, and `group_by.replace_nulls`. I also added a version of `replace_nulls` that java was missing. It is still not decided exactly how we are going to support running windows, but I thought I should get these in in case we do want to use them.

jlowe commented on java/src/main/java/ai/rapids/cudf/ColumnView.java:
```diff
@@ -1317,16 +1321,49 @@ public final ColumnVector rollingWindow(RollingAggregation op, WindowOptions opt
-   * Output values 1, 3, 6
-   * This currently only works for long values that are not nullable as this is currently a
-   * very simple implementation. It may be expanded in the future if needed.
+   * This is just a convenience method for an inclusive scan ona SUM.
```
```suggestion
   * This is just a convenience method for an inclusive scan on a SUM.
```

jlowe commented on java/src/main/java/ai/rapids/cudf/ScanType.java:
```diff
@@ -0,0 +1,33 @@
+package ai.rapids.cudf;
+
+/*
+ * This is analogous to the native 'scan_type'.
```
Nit: It would be nice to define the semantics of inclusive vs. exclusive here.

jlowe commented on java/src/main/java/ai/rapids/cudf/ReplacePolicy.java:
```diff
@@ -0,0 +1,40 @@
+package ai.rapids.cudf;
+
+/*
+ * This is analogous to the native 'replace_policy'.
```
Nit: It would be nice to define the semantics of the values here for the Java devs that don't want to dive into the native code.

jlowe reviewed (approved):
Nice cleanup for prefix sum!  Looks good to me, just minor nits.

revans2 commented:
@jlowe I think I have addressed all of your review comments, and updated the comments for a few other Enums too, because I was following a pattern that was already there.

revans2 commented:
@gpucibot merge
